tensorizer/serialization.py

```diff
@@ -609,7 +574,18 @@ def read_tensors(
                 # Read the dtype of the tensor.
                 dtype_len = struct.unpack("<B", headers[idx : idx + 1])[0]
                 dtype_end = idx + dtype_len + 1
-                dtype = headers[idx + 1 : dtype_end]
+                dtype = headers[idx + 1: dtype_end].decode("utf-8")
+
+                numpy_dtype, *torch_dtype = dtype.split(OPAQUE_DTYPE_SEP)
+                if len(torch_dtype) == 0:
+                    torch_dtype = None
+                elif len(torch_dtype) == 1:
+                    torch_dtype = torch_dtype[0]
+                else:
+                    raise ValueError("Can't deserialize a tensor with "
+                                     "multiple opaque dtype separators "
+                                     f"({OPAQUE_DTYPE_SEP}) in its dtype: "
+                                     f"{dtype}")
```

Comment:
Including a literal null byte in a string meant to be read by a human might lead to weird things. We should use its `repr` instead so it formats as `'\x00'`:
```suggestion
                                     f"({OPAQUE_DTYPE_SEP!r}) in its dtype: "
                                     f"{dtype!r}")
```